Fix rare crash when font is specified by filename

Darkyenus opened the pull request:
Fixes #604, though I am not sure if this is a correct/complete fix, as discussed in the issue.
In any case, it shouldn't break anything.

Even if this doesn't get merged, it could be useful for others facing the same issue.

alerque commented on core/base-shaper.lua:
```diff
@@ -29,6 +29,7 @@ SILE.shapers.base = std.object {
       SILE.settings.temporarily(function()
         SILE.settings.set("font.size", options.size)
         SILE.settings.set("font.family", options.family)
+        if options.filename then SILE.settings.set("font.filename", options.filename) end
```
Why does this like need to be wrapped in an `if` statement and the preceding ones do not?

Darkyenus replied:
Short answer: Because it crashed without it.

Long answer: I believe it has to do with the fact, that when the font is specified by filename here, the font family is specified by empty string. But when it is specified by the family, the filename is `nil`, which crashes the `set` function.

As to why that is, I don't know. I don't have the time right now to understand this part of the SILE, so I stopped working on this when I got this fix/workaround.

alerque replied:
I wonder if #607 will fix the issue with `set()`.

alerque replied:
I have verified both that your fix is needed and that the changes in the earlier PR fix the settings so that a nil value does not cause a crash. This can be simply:

```suggestion
        SILE.settings.set("font.filename", options.filename)
```

alerque commented:
I'm sorry it took so long to get around to reviewing this properly. In the mean time the PR needs an update to resolve the merge conflict in addition to the tweak that I mentioned in the review comment. If you can address those I'd be happy to get this merged. If for some reason you don't have time I'm also happy to fix it up if you check off the box allowing maintainers to push to this branch, or I can cherry pick the fix in another PR. Just let me know. And thanks again...

Darkyenus commented:
Thanks for getting around to this. I have rebased and removed the `if`, as you suggested. Since I haven't touched sile for over half a year, I haven't re-tested the fix, but it should be ok.

alerque commented:
Thanks for cleaning it up for merge, even if you haven't been using it! By the way I make good use of the stack trace features you added about 6 days a week.
